ldms/src/ldmsd/ldmsd_request.c

```diff
@@ -5050,7 +5050,7 @@ static int __greeting_path_req_handler(ldmsd_req_ctxt_t reqc)
 		attr.attr_len = strlen(myself);
 		ldmsd_hton_req_attr(&attr);
 		ldmsd_append_reply(reqc, (char *)&attr, sizeof(attr), LDMSD_REQ_SOM_F);
-		ldmsd_append_reply(reqc, myself, strlen(myself), 0);
+		ldmsd_append_reply(reqc, myself, strlen(myself)+1, 0);
```

Comment:
Why do you +1 here?

Reply:
so that the content is terminated by `'\0'`. Otherwise, the next level ldmsd will have to add `'\0'` before adding its name to the reply.

Reply:
pls. make certain that the ldms_ctxt_alloc takes that into account.

Reply:
I'm confused. This is in `greeting` request message handler in ldmsd.